src/foundation.cs

```diff
@@ -2568,6 +2592,620 @@ interface NSMetadataQuery {
+		[Field ("NSMetadataItemAppleLoopsRootKeyKey")]
+		NSString AppleLoopsRootKeyKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemAppleLoopsKeyFilterTypeKey")]
+		NSString AppleLoopsKeyFilterTypeKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemAppleLoopsLoopModeKey")]
+		NSString AppleLoopsLoopModeKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemAppleLoopDescriptorsKey")]
+		NSString AppleLoopDescriptorsKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemMusicalInstrumentCategoryKey")]
+		NSString MusicalInstrumentCategoryKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemMusicalInstrumentNameKey")]
+		NSString MusicalInstrumentNameKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemCFBundleIdentifierKey")]
+		NSString CFBundleIdentifierKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemInformationKey")]
+		NSString InformationKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemDirectorKey")]
+		NSString DirectorKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemProducerKey")]
+		NSString ProducerKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemGenreKey")]
+		NSString GenreKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemPerformersKey")]
+		NSString PerformersKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemOriginalFormatKey")]
+		NSString OriginalFormatKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemOriginalSourceKey")]
+		NSString OriginalSourceKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemAuthorEmailAddressesKey")]
+		NSString AuthorEmailAddressesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemRecipientEmailAddressesKey")]
+		NSString RecipientEmailAddressesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemAuthorAddressesKey")]
+		NSString AuthorAddressesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemRecipientAddressesKey")]
+		NSString RecipientAddressesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemIsLikelyJunkKey")]
+		NSString IsLikelyJunkKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemExecutableArchitecturesKey")]
+		NSString ExecutableArchitecturesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemExecutablePlatformKey")]
+		NSString ExecutablePlatformKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemApplicationCategoriesKey")]
+		NSString ApplicationCategoriesKey { get; }
+
+		[NoWatch, NoTV, NoiOS, Mac (10, 9)]
+		[Field ("NSMetadataItemIsApplicationManagedKey")]
+		NSString IsApplicationManagedKey { get; }
```

Comment:
Can all this Keys be in a smart enum? Will make it better for the API and safer.